src/function/scalar/system/aggregate_export.cpp

```diff
@@ -0,0 +1,286 @@
+	input.data[1].Orrify(input.size(), state1_data);
+
+	auto result_ptr = FlatVector::GetData<string_t>(result);
+
+	for (idx_t i = 0; i < input.size(); i++) {
+		auto state0_idx = state0_data.sel->get_index(i);
+		auto state1_idx = state1_data.sel->get_index(i);
+
+		auto &state0 = ((string_t *)state0_data.data)[state0_idx];
+		auto &state1 = ((string_t *)state1_data.data)[state1_idx];
+
+		// if both are NULL, we return NULL. If either of them is not, the result is that one
+		if (!state0_data.validity.RowIsValid(state0_idx) && !state1_data.validity.RowIsValid(state1_idx)) {
+			FlatVector::SetNull(result, i, true);
+			continue;
+		}
+		if (state0_data.validity.RowIsValid(state0_idx) && !state1_data.validity.RowIsValid(state1_idx)) {
+			result_ptr[i] =
+			    StringVector::AddStringOrBlob(result, (const char *)state0.GetDataUnsafe(), bind_data.state_size);
+			continue;
+		}
+		if (!state0_data.validity.RowIsValid(state0_idx) && state1_data.validity.RowIsValid(state1_idx)) {
+			result_ptr[i] =
+			    StringVector::AddStringOrBlob(result, (const char *)state1.GetDataUnsafe(), bind_data.state_size);
+			continue;
+		}
+
+		// we actually have to combine
+		D_ASSERT(state0.GetSize() == bind_data.state_size);
+		D_ASSERT(state1.GetSize() == bind_data.state_size);
+
+		memcpy(local_state.state_buffer0.get(), state0.GetDataUnsafe(), bind_data.state_size);
+		memcpy(local_state.state_buffer1.get(), state1.GetDataUnsafe(), bind_data.state_size);
+
+		bind_data.aggr.combine(local_state.state_vector0, local_state.state_vector1, 1);
+
+		result_ptr[i] =
+		    StringVector::AddStringOrBlob(result, (const char *)local_state.state_buffer1.get(), bind_data.state_size);
+	}
+}
+
+static unique_ptr<FunctionData> BindAggregateState(ClientContext &context, ScalarFunction &bound_function,
+                                                   vector<unique_ptr<Expression>> &arguments) {
+
+	// grab the aggregate type and bind the aggregate again
+
+	// the aggregate name and types are in the logical type of the aggregate state, make sure its sane
+	auto &arg_return_type = arguments[0]->return_type;
+
+	if (arg_return_type.id() != LogicalTypeId::AGGREGATE_STATE) {
+		throw BinderException("Can only FINALIZE aggregate state, not %s", arg_return_type.ToString());
+	}
+	// combine
+	if (arguments.size() == 2 && arguments[0]->return_type != arguments[1]->return_type) {
+		throw BinderException("Cannot COMBINE aggregate states from different functions, %s <> %s",
+		                      arguments[0]->return_type.ToString(), arguments[1]->return_type.ToString());
+	}
+
+	// following error states are only reachable when someone messes up creating the state_type which is impossible from
+	// SQL
+
+	auto state_type = AggregateStateType::GetStateType(arg_return_type);
+
+	// now we can look up the function in the catalog again and bind it
+	auto func = Catalog::GetCatalog(context).GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, DEFAULT_SCHEMA,
```

Comment:
Don't bound functions have the schema attached?

Reply:
Yes but its the wrong function, we just bound EXPORT/COMBINE and not the aggregate that was used to create the state